tools/circt-verilog/circt-verilog.cpp

```diff
@@ -16,22 +16,293 @@
+      cl::Prefix, cl::cat(cat)};
+
+  cl::opt<uint32_t> errorLimit{
+      "error-limit",
+      cl::desc("Limit on the number of errors that will be printed. Setting "
+               "this to zero will disable the limit."),
+      cl::value_desc("limit"), cl::cat(cat)};
+
+  cl::list<std::string> suppressWarningsPaths{
+      "suppress-warnings",
+      cl::desc("One or more paths in which to suppress warnings"),
+      cl::value_desc("filename"), cl::cat(cat)};
+
+  //===--------------------------------------------------------------------===//
+  // File lists
+  //===--------------------------------------------------------------------===//
+
+  cl::opt<bool> singleUnit{
+      "single-unit",
+      cl::desc("Treat all input files as a single compilation unit"),
+      cl::init(false), cl::cat(cat)};
+
+  cl::list<std::string> libraryFiles{
+      "l",
+      cl::desc(
+          "One or more library files, which are separate compilation units "
+          "where modules are not automatically instantiated."),
+      cl::value_desc("filename"), cl::Prefix, cl::cat(cat)};
+};
+} // namespace
+
+static CLOptions opts;
+
 //===----------------------------------------------------------------------===//
 // Implementation
 //===----------------------------------------------------------------------===//
 
-static LogicalResult execute(MLIRContext *context) {
-  // This is where we would call out to ImportVerilog to parse the input.
+static LogicalResult executeWithSources(MLIRContext *context,
+                                        llvm::SourceMgr &sourceMgr) {
+  // Create the timing manager we use to sample execution times.
+  DefaultTimingManager tm;
+  applyDefaultTimingManagerCLOptions(tm);
+  auto ts = tm.getRootScope();
+
+  // Map the command line options to `ImportVerilog`'s conversion options.
+  ImportVerilogOptions options;
+  options.onlyLint = opts.onlyLint;
+  options.onlyParse = opts.onlyParse;
+
+  options.includeDirs = opts.includeDirs;
+  options.includeSystemDirs = opts.includeSystemDirs;
+  options.libDirs = opts.libDirs;
+  options.libExts = opts.libExts;
+  options.excludeExts = opts.excludeExts;
+
+  options.defines = opts.defines;
+  options.undefines = opts.undefines;
+  if (opts.maxIncludeDepth.getNumOccurrences() > 0)
+    options.maxIncludeDepth = opts.maxIncludeDepth;
+  options.librariesInheritMacros = opts.librariesInheritMacros;
+
+  if (opts.timeScale.getNumOccurrences() > 0)
+    options.timeScale = opts.timeScale;
+  options.allowUseBeforeDeclare = opts.allowUseBeforeDeclare;
+  options.ignoreUnknownModules = opts.ignoreUnknownModules;
+  if (!opts.onlyLint)
+    options.topModules = opts.topModules;
+  options.paramOverrides = opts.paramOverrides;
+
+  options.warningOptions = opts.warningOptions;
+  if (opts.errorLimit.getNumOccurrences() > 0)
+    options.errorLimit = opts.errorLimit;
+  options.suppressWarningsPaths = opts.suppressWarningsPaths;
+
+  options.singleUnit = opts.singleUnit;
```

Comment:
bool -> optional seems smidge weird but this makes sense I think.
If can check whether it was specified at all, maybe only set it then? Not sure what slang's version is optional shrug.
(or have `-single-unit` `ValueDisallowed` or something like that re:it only indicates true or something...?)

Reply:
I think Slang tries to explicitly capture whether the option was specified or not, in order to provide override-to-true, override-to-false, and default options. My thinking here was to always set this to a known value (since it's exposed as a command line arg of `circt-verilog`), instead of catering to that third option. But `ImportVerilog` exposes the optionality because the underlying Slang thing is optional.